packages/web-components/src/drawer/drawer.spec.ts

```diff
@@ -0,0 +1,276 @@
+
+  test('show and hide methods should toggle open state of drawer', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+          <fluent-drawer></fluent-drawer>
+        `;
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+
+    await element.evaluate((node: Drawer) => {
+      node.show();
+    });
+
+    await expect(element).toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', true);
+
+    await element.evaluate((node: Drawer) => {
+      node.close();
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+  });
+
+  test('hide method should close drawer', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-drawer open></fluent-drawer>
+          `;
+    });
+
+    await expect(element).toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', true);
+
+    await element.evaluate((node: Drawer) => {
+      node.close();
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+  });
+
+  test('show method should open drawer', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+              <fluent-drawer ></fluent-drawer>
+            `;
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+
+    await element.evaluate((node: Drawer) => {
+      node.show();
+    });
+
+    await expect(element).toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', true);
+  });
+
+  test('should emit an event when open property changes', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-drawer></fluent-drawer>
+        `;
+    });
+
+    const [wasOpened] = await Promise.all([
+      element.evaluate(
+        node =>
+          new Promise(resolve => {
+            node.addEventListener('onOpenChange', () => resolve(true));
+          }),
+      ),
+      await element.evaluate((node: Drawer) => {
+        node.show();
+      }),
+    ]);
+
+    expect(wasOpened).toBe(true);
+  });
+
+  test('drawer should close on escape keypress', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-drawer open tabindex="0"></fluent-drawer>
+        `;
+    });
+
+    await expect(element).toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', true);
+
+    await element.evaluate(node => {
+      node.focus();
+    });
+
+    await expect(element).toBeFocused();
+
+    await element.press(keyEscape);
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+  });
```

Comment:
Can you test tab key handling and the expected results whether focus is trapped or not?